simapp/simd/cmd/testnet_test.go

```diff
@@ -13,14 +13,17 @@ import (
 	"github.com/cosmos/cosmos-sdk/client/flags"
 	"github.com/cosmos/cosmos-sdk/server"
 	"github.com/cosmos/cosmos-sdk/simapp"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/cosmos/cosmos-sdk/x/auth"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 	genutiltest "github.com/cosmos/cosmos-sdk/x/genutil/client/testutil"
 	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
 )
 
 func Test_TestnetCmd(t *testing.T) {
 	home := t.TempDir()
-	encodingConfig := simapp.MakeTestEncodingConfig()
+	encodingConfig := moduletestutil.MakeTestEncodingConfig(staking.AppModuleBasic{}, auth.AppModuleBasic{})
```

Comment:
- We marshal the genesis in `initGenFiles` so we need auth (https://github.com/cosmos/cosmos-sdk/blob/main/simapp/simd/cmd/testnet.go#L370)
- Then [here](https://github.com/cosmos/cosmos-sdk/blob/5decd22cd3a9259bbae151e1e283ab756c4323c7/simapp/simd/cmd/testnet.go#L325) we need `staking` for encoding our validator msg created [there](https://github.com/cosmos/cosmos-sdk/blob/5decd22cd3a9259bbae151e1e283ab756c4323c7/simapp/simd/cmd/testnet.go#L295). Otherwise, we get `unable to resolve type URL /cosmos.staking.v1beta1.MsgCreateValidator`